mdl/src/main/cft/mdlElasticsearch.yml

```diff
@@ -81,7 +81,100 @@ Conditions:
+          - !Ref ReleaseVersion
+          - 'es-lambda-role'
+      Policies:
+        - PolicyName: root
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - iam:*
+                Resource: '*'
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogGroup
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                Resource: arn:aws:logs:*:*:*
+  ManageServiceLinkedRoleFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Handler: index.handler
+      Role: !GetAtt ManageESLambdaExecutionRole.Arn
+      Code:
+        ZipFile: !Sub |
+          import boto3
+          from botocore.exceptions import ClientError
+          import cfnresponse
+          from time import sleep
+
+          def handler(event, context):
+              if event['RequestType'] == 'Create':
+                  try:
+                      iam_client = boto3.client('iam')
+                      response = iam_client.get_role(
+                          RoleName='AWSServiceRoleForAmazonElasticsearchService'
+                      )
+                      arn = response.get('Role', {}).get('Arn')
+                      print('INFO: Role exists, nothing to do. ARN: {}'.format(arn))
+                      response_data = {'Message': 'Role exists, nothing to do. Service response: {}.'.format(response)}
+                      cfnresponse.send(event, context, cfnresponse.SUCCESS, response_data, 'ServiceLinkedRoleFunction')
+                  except ClientError as e:
+                      if e.response['Error']['Code'] == 'NoSuchEntity':
+                          __create_service_linked_role(event, context)
+                      else:
+                          print('ERROR: IAM service returned an Error. Exception: {}'.format(e))
+                          response_data = {'Message': 'IAM service returned an Error. Exception: {}'.format(e)}
+                          cfnresponse.send(event, context, cfnresponse.FAILED, response_data, 'ServiceLinkedRoleFunction')
+
+              elif event['RequestType'] == 'Delete':
+                  response_data = {'Message': 'Deleting custom resource.'}
+                  cfnresponse.send(event, context, cfnresponse.SUCCESS, response_data, 'ServiceLinkedRoleFunction')
+
+          def __create_service_linked_role(event, context):
+              try:
+                  iam_client = boto3.client('iam')
+                  print('INFO: Creating service-linked role for VPC Elasticsearch.')
+                  response = iam_client.create_service_linked_role(
+                      AWSServiceName='es.amazonaws.com',
+                      Description='service-linked role for VPC Elasticsearch domain.'
+                  )
+                  print('INFO: Role created. Service response: {}'.format(response))
+                  sleep(180)
```

Comment:
A comment would be useful here explaining why a sleep is required.

Reply:
Yeah, thanks for pointing that out. I will add comments and update this PR.